specifyweb/businessrules/groupperson_rules.py

```diff
@@ -6,7 +6,11 @@
 @orm_signal_handler('pre_save', 'Groupperson')
 def agent_cannot_be_in_self(groupperson):
     if groupperson.member_id == groupperson.group_id:
-        raise BusinessRuleException('a group cannot be made a member of itself')
+        raise BusinessRuleException(
+            'a group cannot be made a member of itself', 
+            {"table" : "GroupPerson",
```

Comment:
Whenever possible, add a `fieldName` to the response payload. That way front-end can attribute error to a single field. In cases when fieldName is missing, front-end will display a dialog

Reply:
Added in 93b1277. 
I did not yet add 'fieldName' to TreeBusinessRule Exceptions. Most of the time the error can not be easily distilled to a single field. We can probably separate the frontend implementation of TreeBusinessRule Exceptions from BusinessRule Exceptions to provide enough information in 'form-error form' so the user can resolve the issues and does not see the error dialog